spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/TableWriter.scala

```diff
@@ -79,8 +95,11 @@ class TableWriter[T] private (
   private val isCounterUpdate =
     tableDef.columns.exists(_.isCounterColumn)
 
+  private val containsCollectionBehaviors =
+    columnSelector.collect{ case cn:CollectionColumnName => cn }.nonEmpty
```

Comment:
There is nothing wrong with isInstanceOf where you'd have to do pattern matching.